### ISSUE TITLE
lwt 2.5.0: doesn't build with musl-libc

### ISSUE THREAD

**Issue by edwintorok** [ocsigen/lwt#175]
I tried building lwt 2.5.0 with musl-gcc on Debian, but it failed:
```
# + /home/edwin/.opam/4.02.1+musl/bin/ocamlfind ocamlopt -linkpkg -g -I src/unix -linkpkg -package unix -package bytes -package bigarray -package camlp4.quotations.o -package camlp4.extend -package camlp4 syntax/lwt-syntax.cmxa src/core/lwt.cmxa src/logger/lwt-log.cmxa src/unix/lwt-unix.cmxa examples/unix/logging.cmx -o examples/unix/logging.native
# src/unix/liblwt-unix_stubs.a(lwt_unix_stubs.o): In function `worker_getprotobyname':
# lwt_unix_stubs.c:(.text+0x26c3): undefined reference to `getprotobyname_r'
# src/unix/liblwt-unix_stubs.a(lwt_unix_stubs.o): In function `worker_getprotobynumber':
# lwt_unix_stubs.c:(.text+0x2727): undefined reference to `getprotobynumber_r'
```

Looks like `netdb.h` on musl libc does have reentrant `gethostbyaddr`, `gethostbyname`, `getservbyname` and `getservbyaddr`, but there are no reentrant version of `getprotobyname` and `getprotobynumber` in musl libc 1.1.5 and musl libc 1.1.10 either, and `discover.ml` only tests for the former.

To reproduce on Debian jessie:
```
# workaround for discover.ml adding wrong pthread.h (from glibc not musl)
$ sudo apt-get install musl-dev musl-tools
$ export PTHREAD_CFLAGS=
$ export PTHREAD_LIBS=-lpthread
$ opam switch 4.02.1+musl
$ eval `opam config env`
$ opam install lwt.2.5.0
```

**Comment by vbmithr** [comment]
Is this fixed with 2.5.1? Should be, please try.

**Comment by aantron** [comment]
I was unable to reproduce the original failure (on Ubuntu 14.04 though). It seems this should have been fixed by #191 (commit https://github.com/ocsigen/lwt/commit/cb1c691d12bbd7625d83faec1b1a68c1d91258a7), which was included in `2.5.1`. Please reopen and let me know if not.
